### PR TITLE
Fallback to legacy query optimizer for queries over replicated table

### PR DESCRIPTION
**Pull request by pengzhout**

### CODE REVIEW

File: libnaucrates/src/md/IMDRelation.cpp [hunk 1]
```diff
@@ -40,6 +41,10 @@ IMDRelation::PstrDistrPolicy
 		case EreldistrRandom:
 			return CDXLTokens::PstrToken(EdxltokenRelDistrRandom);
 		default:
+			GPOS_RAISE(
```
Comment (vraghavan78) [inline libnaucrates/src/md/IMDRelation.cpp]: i guess. but instead why not do it inside GPDB?

Comment (bhuvnesh2703) [reply libnaucrates/src/md/IMDRelation.cpp]: May be look at `CTranslatorRelcacheToDXL::Pmdrel()`, it calls `ereldistribution = Ereldistribution(pgppolicy);`. You could change the caller i.e `Pmdrel()` to handle Sentinel distribution policy and raise an exception. 
In debug build, currently it asserts in `Ereldistribution` so it falls back earlier than the place where you have added the exception.
In retail build, it goes to create a new object of `CMDRelationGPDB` and raises the exception added by you.
bC & EK

Comment (pengzhout) [reply libnaucrates/src/md/IMDRelation.cpp]: Ahh, found the place to fallback inside GPDB:

diff --git a/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp b/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp
index 0393c5e..3359033 100644
--- a/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp
+++ b/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp
@@ -922,7 +922,7 @@ CTranslatorRelcacheToDXL::Ereldistribution
                return IMDRelation::EreldistrMasterOnly;
        }

-       GPOS_ASSERT(!"Unrecognized distribution policy");
+       GPOS_RAISE(gpdxl::ExmaMD, ExmiDXLUnrecognizedType, GPOS_WSZ_LIT("unrecognized distribution policy"));
        return IMDRelation::EreldistrSentinel;
 }

Am I right? @bhuvnesh2703

Comment (pengzhout) [reply libnaucrates/src/md/IMDRelation.cpp]: @vraghavan78 @bhuvnesh2703 thanks, will close this one.

**Pull request closed without merging**